crates/oxc_linter/src/rules/eslint/eq_eq_eq.rs

```diff
@@ -34,62 +74,170 @@ declare_oxc_lint!(
-    match operator {
-        BinaryOperator::Equality => BinaryOperator::StrictEquality,
-        BinaryOperator::Inequality => BinaryOperator::StrictInequality,
-        _ => unreachable!(),
+impl Rule for EqEqEq {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let obj1 = value.get(0);
+        let obj2 = value.get(1);
+
+        Self {
+            compare_type: obj1
+                .and_then(serde_json::Value::as_str)
+                .map(CompareType::from)
+                .unwrap_or_default(),
+            null_type: obj2
+                .and_then(|v| v.get("null"))
+                .and_then(serde_json::Value::as_str)
+                .map(NullType::from)
+                .unwrap_or_default(),
+        }
     }
-}
 
-impl Rule for EqEqEq {
     fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
         let AstKind::BinaryExpression(binary_expr) = node.kind() else { return };
+        let is_null = is_null_check(binary_expr);
+        let enforce_rule_for_null = matches!(self.null_type, NullType::Always);
+        let enforce_inverse_rule_for_null = matches!(self.null_type, NullType::Never);
+
         if !matches!(binary_expr.operator, BinaryOperator::Equality | BinaryOperator::Inequality) {
+            if enforce_inverse_rule_for_null && is_null {
+                let operator = binary_expr.operator.as_str();
+                // There are some uncontrolled cases to auto fix.
+                // In ESlint, `null >= null` will be auto fixed to `null > null` which is also wrong.
+                // So I just report it.
+                ctx.diagnostic(EqEqEqDiagnostic(
+                    operator,
+                    &operator[0..operator.len() - 1],
+                    binary_expr.span,
+                ));
+            }
+
             return;
         }
 
-        let is_valid_comparison = match (&binary_expr.left, &binary_expr.right) {
-            (Expression::UnaryExpression(unary_expr), _)
-            | (_, Expression::UnaryExpression(unary_expr)) => {
-                matches!(unary_expr.operator, UnaryOperator::Typeof)
-            }
-            (lhs, rhs) => {
-                lhs.is_null()
-                    || rhs.is_null()
-                    || (lhs.is_literal_expression() && rhs.is_literal_expression())
-            }
-        };
+        let is_type_of_binary_bool = is_type_of_binary(binary_expr);
+        let are_literals_and_same_type_bool =
+            are_literals_and_same_type(&binary_expr.left, &binary_expr.right);
+        // The "smart" option enforces the use of `===` and `!==` except for these cases:
+        //  - Comparing two literal values
+        //  - Evaluating the value of typeof
+        //  - Comparing against null
+        if matches!(self.compare_type, CompareType::Smart)
+            && (is_type_of_binary_bool || are_literals_and_same_type_bool || is_null)
+        {
+            return;
+        }
 
-        if !is_valid_comparison {
```

Comment:
I need to run fixer in some situation which contained in `is_valid_comparison == true`. so I refactor this rule.